### ISSUE TITLE
Improve test coverage and create guideline

### ISSUE THREAD

**Issue by russellwmy** [neuronika/neuronika#71]
It maybe a future problem. However, it is better to improve the test coverage of the project and create guideline for adding LOC to the project. The idea solution is utilizing Github Action and build the automated checks.

**Comment by frjnn** [comment]
Yes, test coverage will be greatly improved in the coming update, to be 100% honest we are still uncertain on how optimizers should be tested. Concerning guidelines we will gladly and happily provide guidance and help should you, or anybody else, want to open an issue. Nevertheless, you have two very good points.
